File: strings/valid_palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if not s.strip():
            return True

        l = 0
        r = len(s) - 1

        while r >= l:
            if s[l].isalnum() and s[r].isalnum():
                if s[l].lower() == s[r].lower():
                    l += 1
                    r -= 1
                    continue
                else:
                    return False

            else:
                if not s[l].isalnum():
                    l += 1
                
                if not s[r].isalnum():
                    r -= 1

        return True

File: strings/valid_palindrome.rb
def is_palindrome(s)
    return true if s.strip.empty?
    
    l = 0
    r = s.length - 1
    
    while r >= l
        if s[l].match?(/[[:alnum:]]/) && s[r].match?(/[[:alnum:]]/)
            if s[l].downcase == s[r].downcase
                l += 1
                r -= 1
                next
            else
                return false
            end
        else
            l += 1 if !s[l].match?(/[[:alnum:]]/)
            r -= 1 if !s[r].match?(/[[:alnum:]]/)
        end
    end
    
    true
end